src/crh/pedersen/mod.rs

```diff
@@ -133,28 +138,30 @@ impl<C: ProjectiveCurve, W: Window> TwoToOneCRH for CRH<C, W> {
     /// A simple implementation method: just concat the left input and right input together
     ///
     /// `evaluate` requires that `left_input` and `right_input` are of equal length.
-    fn evaluate(
+    fn compress<T: Borrow<Self::Output>>(
         parameters: &Self::Parameters,
-        left_input: &[u8],
-        right_input: &[u8],
+        left_input: T,
+        right_input: T,
     ) -> Result<Self::Output, Error> {
+        let left_input_bytes = ark_ff::to_bytes!(left_input.borrow())?;
+        let right_input_bytes = ark_ff::to_bytes!(right_input.borrow())?;
```

Comment:
Let's add a TODO here to revisit and remove this when we deprecate `to_bytes!`

Reply:
all `to_bytes!` in CRH and merkle tree has been replaced by `serialize_unchecked` (assuming `serialize_unchecked` is consistent with `to_non_unique_bytes` in constraint world)